mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -92,9 +91,112 @@ def _set_lot_number_propagation_data_from_bom(self):
             else:
                 propagate_move.propagate_lot_number = True
 
+    def _get_quantity_produced_issues(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return False
+        return super()._get_quantity_produced_issues()
+
+    def _set_quantities(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return True
+        return super()._set_quantities()
+
+    def _auto_production_checks(self):
+        if (
+            self.env.context.get("bypass_pre_button_mark_done_checks")
+            and self.is_lot_number_propagated
+        ):
+            return True
+        return super()._auto_production_checks()
+
     def pre_button_mark_done(self):
-        self._create_and_assign_propagated_lot_number()
-        return super().pre_button_mark_done()
+        if all(prod._auto_production_checks() for prod in self):
+            self._create_and_assign_propagated_lot_number()
+        res = super().pre_button_mark_done()
+        if isinstance(res, dict) and res["res_model"] == "mrp.batch.produce":
```

Comment:
this part can be maybe split in smaller methods

Reply:
I usually think it's good to split in smaller function, but there it's a single process of multiple small steps/checks, hence why I added lots of comment to explain what each is doing.

IMO splitting this doesn't improve the readability of the code